### PR TITLE
feat :sparkles: : Added Timer + TimerButton + Formater

### PR DESCRIPTION
**Pull request by PrzemekChociaj**

### CODE REVIEW

File: src/components/Features/Formater.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const formatTime = (time: number) => {
+    const minutes = Math.floor(time / 60);
```
Comment (qbawalat) [inline src/components/Features/Formater.tsx]: 60 is magic number - extract to const e.g. MINUTES_IN_ONE_HOUR

File: src/components/Features/TimerButton.tsx [hunk 2]
```diff
@@ -1,12 +1,11 @@
-const TimerButton = ({ setIsRunning, isRunning }) => {
-	const toggleTimer = () => {
-		setIsRunning((prevIsRunning) => !prevIsRunning);
-	};
+const TimerButton = ({ isRunning, toggleTimer }) => {
```
Comment (qbawalat) [inline src/components/Features/TimerButton.tsx]: Missing FunctionComponent<Props> type

File: src/components/Timer/Timer.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useRef } from 'react';
+import TimerButton from '../Features/TimerButton';
+import { formatTime } from '../Features/Formater';
+
+const Timer: React.FunctionComponent = () => {
```
Comment (qbawalat) [inline src/components/Timer/Timer.tsx:5]: 1. Don't import react
2. 2Dont use React.FunctionCOmponent - import FUnctionComponent from React and just write TImer: FunctionComponent

File: src/components/Timer/Timer.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	const stopTimer = () => {
+		if (!isRunning) return;
+
+		setIsRunning(false);
+		if (timerRef.current) {
+			clearInterval(timerRef.current);
+			timerRef.current = null;
+		}
+	};
+
```
Comment (qbawalat) [inline src/components/Timer/Timer.tsx]: extract to custom hook e.g. useTimer

**Review by qbawalat — CHANGES_REQUESTED** [review]

File: src/components/Features/Formater.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const formatTime = (time: number) => {
```
Comment (khali16) [inline src/components/Features/Formater.tsx]: Missing return type

File: src/components/Features/TimerButton.tsx [hunk 6]
```diff
@@ -1,12 +1,11 @@
...
 	return (
 		<button
-			className={`px-4 py-2 rounded-md ${
-				isRunning ? 'bg-red-500' : 'bg-green-500'
-			} text-white`}
+			className={`${
+				isRunning
```
Comment (khali16) [inline src/components/Features/TimerButton.tsx]: Conditional styles should be extracted to a separate function

**Review by khali16 — CHANGES_REQUESTED** [review]

File: src/components/Features/TimerButton.tsx [hunk 7]
```diff
@@ -1,11 +1,18 @@
-const TimerButton = ({ isRunning, toggleTimer }) => {
+const ButtonClassName = (isRunning: boolean): string => {
```
Comment (khali16) [inline src/components/Features/TimerButton.tsx]: - It is a function so you should change naming convention - e.g. getButtonColor;
- no need for bracers and return statement - use oneliner function

**Pull request merged**